### PR TITLE
 [Typing][B-62] Add type annotations for `python/paddle/text/datasets/imikolov.py`

### PR DESCRIPTION
**Pull request by enkilee**
### PR Category

User Experience

### PR Types

Improvements

### Description

- https://github.com/PaddlePaddle/Paddle/issues/65008
B62

### CODE REVIEW

File: python/paddle/text/datasets/imikolov.py [hunk 1]
```diff
@@ -81,14 +88,22 @@ class Imikolov(Dataset):
 
     """
 
+    data_file: str | None
+    data_type: _ImikolovDataType
+    window_size: int
+    mode: _ImikolovDataSetMode
+    min_word_freq: int
+    download: bool
```
Comment (megemini) [inline python/paddle/text/datasets/imikolov.py]:
```suggestion
```

貌似没有这个属性？

File: python/paddle/text/datasets/imikolov.py [hunk 2]
```diff
@@ -81,14 +88,22 @@ class Imikolov(Dataset):
...
-        mode='train',
-        min_word_freq=50,
-        download=True,
+        data_file: str | None = None,
+        data_type: _ImikolovDataType = 'NGRAM',
+        window_size: int = -1,
+        mode: _ImikolovDataSetMode = 'train',
+        min_word_freq: int = 50,
+        download: bool = True,
     ):
```
Comment (megemini) [inline python/paddle/text/datasets/imikolov.py]:
```suggestion
    ) -> None:
```

File: python/paddle/text/datasets/imikolov.py [hunk 3]
```diff
@@ -180,8 +195,8 @@ def _load_anno(self):
                 else:
                     raise AssertionError('Unknow data type')
 
-    def __getitem__(self, idx):
+    def __getitem__(self, idx: int) -> tuple[npt.NDArray[np.int_]]:
```
Comment (megemini) [inline python/paddle/text/datasets/imikolov.py]:
```suggestion
    def __getitem__(self, idx: int) -> tuple[npt.NDArray[np.int_], npt.NDArray[np.int_]:
```

返回的是 `(src_seq, trg_seq)` ?

Comment (enkilee) [reply python/paddle/text/datasets/imikolov.py]: 恩，对，看漏了一个。

File: python/paddle/text/datasets/imikolov.py (hunk 2 above)
Comment (enkilee) [reply python/paddle/text/datasets/imikolov.py]: 收到

File: python/paddle/text/datasets/imikolov.py (hunk 1 above)
Comment (enkilee) [reply python/paddle/text/datasets/imikolov.py]: 收到

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**